### PR TITLE
feat: add new keybinding "b" to run a command in the focused node

### PR DESCRIPTION
**Pull request by be-ndee**
Press "b" when you are in the tree. User can enter a shell command like "ls -la". That command is then run in the directory on which your cursor is, at the moment.

### CODE REVIEW

**Comment by be-ndee** [comment]
Thanks for the idea, sounds good to me! But what do you mean with "you don't have to implement running the command yourself."?

**Comment by Gelio** [comment]
Thanks!

> you don't have to implement running the command yourself.

I mean [this part](https://github.com/kyazdani42/nvim-tree.lua/pull/1024/files#diff-5833be1c49e26015af8e5a920739c0fd030c363a048f7abaf221c6559e29a178R25-R32):

image

Vim will take care of running the shell command if you prefix it with `!` (see `:h :!`)

**Comment by kyazdani42** [comment]
I can make a quick review here by first saying:
1. gelio is right, you don't need the custom code for running a shell command
2. gelio idea is great
3. just remove the comments please :)

**Comment by be-ndee** [comment]
@Gelio thanks, I will have a look at that.

@kyazdani42 will do that. I've though some comments would make the code easier to understand :)

**Comment by Gelio** [comment]
> Do you know how to do that? I did not find anything to enter the command-line. I just found vim.cmd(), which executes a command. But what we want to do is to only enter the command-line with a predefined command.

Looking at the result of `:verbose map .` in vim-fugitive, I found out how to do it from Lua:

```lua
local path_to_file = "index.html"
vim.api.nvim_input(": " .. path_to_file .. "<Home>")

```

Now you only need to set the `path_to_file` to a real value.

**Comment by be-ndee** [comment]
@Gelio thank you! That works perfectly!

I created to different keybindings:

`b` -> enters the command line with the directoryname
`B` -> enters command line with the file name

So if your cursor is on a file, than you could use `b` to run a command with the directory name or use `B` for the filename.

**Comment by be-ndee** [comment]
@kyazdani42 I think this should be finished now.

File: lua/nvim-tree/actions/run-command.lua
Comment (Gelio) [inline lua/nvim-tree/actions/run-command.lua:7]: I'm not sure if we need a dedicated `run_directory_command`. It seems to correspond to `Pb` (`P` to go to the parent node and then `b`

Removing it would lower the complexity of this function a bit.

I'm not the maintainer, so don't take my word for it :smile: I'm just stating my point of view

File: lua/nvim-tree/actions/run-command.lua
Comment (Gelio) [inline lua/nvim-tree/actions/run-command.lua:10]: Clever! I suppose this method works for both Windows and Linux/OSX paths :+1:

File: README.md
Comment (Gelio) [inline README.md:224]: Minor:

Looks like we are missing an explanation for `B`

Comment (Gelio) [inline README.md:224]: Minor, typo and a slight reword:

```suggestion
- `b` will prompt the user to enter a vim command which is run on the directory of the focused node (where the cursor is)
```

**Review by Gelio — COMMENTED** [review]
Looks promising :+1:

File: lua/nvim-tree/actions/run-command.lua
Comment (be-ndee) [reply lua/nvim-tree/actions/run-command.lua]: Yepp, I removed it :) So we end with a function containing one line of code :-D

File: README.md
Comment (Gelio) [inline README.md:290]: Minor:

I know it's late and it's another round of review, but personally I would rather have this mapping be under `.` to match vim-fugitive :smile: What do you think?

**Review by Gelio — APPROVED** [review]
LGTM! I would only change the mapping to be under `.` instead of `b` to match vim-fugitive, but I understand some people may not use that plugin

File: lua/nvim-tree/actions/run-command.lua
Comment (Gelio) [reply lua/nvim-tree/actions/run-command.lua]: Nice, simple is beautiful :smile:

File: README.md
Comment (kyazdani42) [reply README.md:290]: I agree it would make more sense. `b` is not really a good default for `run command`. I'd have set something like `x` or `e` (execute), but `.` would be ok too.

Comment (be-ndee) [reply README.md:290]: You could achieve that with creating a custom mapping when using nvim-tree :P

`{key = '.', action = 'run_file_command'}`

Comment (kyazdani42) [reply README.md:290]: what we mean here is having a default like this would make more sense :smile:

Comment (be-ndee) [reply README.md]: I was just joking ;) I've changed it to `.`

**Review by Gelio — APPROVED** [review]
Nice! LGTM

I'd only squash the commits into one before merging the PR so the diff is nicer for anyone following commit-by-commit

**Review by kyazdani42 — APPROVED** [review]

**Pull request merged**
